webapi/src/main/scala/org/knora/webapi/responders/v2/OntologyResponderV2.scala

```diff
@@ -1475,209 +1472,196 @@ class OntologyResponderV2(responderData: ResponderData) extends Responder(respon
 
-        _ = if (rdfType != OntologyConstants.Owl.Class.toSmartIri) {
-              throw BadRequestException(s"Invalid rdf:type for property: $rdfType")
+      (newInternalClassDefWithLinkValueProps, cardinalitiesForClassWithInheritance) =
+        OntologyHelpers
+          .checkCardinalitiesBeforeAddingAndIfNecessaryAddLinkValueProperties(
+            internalClassDef = newInternalClassDef,
+            allBaseClassIris = allBaseClassIris.toSet,
+            cacheData = cacheData
+          )
+          .fold(e => throw e.head, v => v)
+
+      // Check that the class definition doesn't refer to any non-shared ontologies in other projects.
+      _ = Cache.checkOntologyReferencesInClassDef(
+            cache = cacheData,
+            classDef = newInternalClassDefWithLinkValueProps,
+            errorFun = { msg: String =>
+              throw BadRequestException(msg)
             }
+          )
 
-        // Check that the class exists.
-
-        ontology = cacheData.ontologies(internalOntologyIri)
-
-        existingClassDef: ClassInfoContentV2 =
-          ontology.classes
-            .getOrElse(
-              internalClassIri,
-              throw BadRequestException(s"Class ${changeCardinalitiesRequest.classInfoContent.classIri} does not exist")
-            )
-            .entityInfoContent
+      // Build the model
+      inheritedCardinalities = cardinalitiesForClassWithInheritance.filterNot { case (propertyIri, _) =>
+                                 newInternalClassDefWithLinkValueProps.directCardinalities.contains(propertyIri)
+                               }
+      propertyIrisOfAllCardinalitiesForClass = cardinalitiesForClassWithInheritance.keySet
+      knoraResourceProperties                = propertyIrisOfAllCardinalitiesForClass.filter(isKnoraResourceProperty(_, cacheData))
+      linkProperties                         = propertyIrisOfAllCardinalitiesForClass.filter(isLinkProp(_, cacheData))
+      linkValueProperties                    = propertyIrisOfAllCardinalitiesForClass.filter(isLinkValueProp(_, cacheData))
+      fileValueProperties                    = propertyIrisOfAllCardinalitiesForClass.filter(isFileValueProp(_, cacheData))
+    } yield ReadClassInfoV2(
+      entityInfoContent = newInternalClassDefWithLinkValueProps,
+      allBaseClasses = allBaseClassIris,
+      isResourceClass = true,
+      canBeInstantiated = true,
+      inheritedCardinalities = inheritedCardinalities,
+      knoraResourceProperties = knoraResourceProperties,
+      linkProperties = linkProperties,
+      linkValueProperties = linkValueProperties,
+      fileValueProperties = fileValueProperties
+    )
+  }
 
-        // Check that the class isn't used in data, and that it has no subclasses.
-        // TODO: If class is used in data, check additionally if the property(ies) being removed is(are) truly used and if not, then allow.
+  private def checkRdfTypeOfClassIsClass(classInfo: ClassInfoContentV2): ClassInfoContentV2 = {
```

Comment:
I'm not sure why this method returns its input. Wouldn't it make more sense to return Unit (as the only relevant action is throwing anyway) or a boolean stating if the check was true or false (although I see that this would somehow mix a "boolean" check with a "throwing" check which might be confusing as well).

Reply:
It is returning if the input is valid, so that I can use it inline:

`    val newClassInfo        = checkRdfTypeOfClassIsClass(request.classInfoContent.toOntologySchema(InternalSchema))
`